plc4j/protocols/s7/src/main/java/org/apache/plc4x/java/s7/netty/Plc4XS7Protocol.java

```diff
@@ -420,22 +423,29 @@ private PlcResponse decodeReadResponse(S7ResponseMessage responseMessage, PlcReq
             FieldItem fieldItem = null;
             ByteBuf data = Unpooled.wrappedBuffer(payloadItem.getData());
             if (responseCode == PlcResponseCode.OK) {
+                // TODO 2018-09-27 jf: array returning only implemented for BOOL, BYTE, INTEGERS, FP
+                // not for CHARS & STRINGS and not for all other bit-strings except for BYTE
                 switch (field.getDataType()) {
                     // -----------------------------------------
                     // Bit
                     // -----------------------------------------
                     case BOOL: {
-                        byte byteValue = data.readByte();
-                        fieldItem = new S7BooleanFieldItem(field.getDataType(),byteValue != 0x00);
+                        Boolean[] booleans = readAllValues(field, i -> data.readByte() != 0x00).toArray(new Boolean[field.getNumElements()]);
```

Comment:
actually sonar finds a presized array as a bug: see here https://stackoverflow.com/a/29444594/850036

Reply:
I'm about to fix it... didn't know that